nomad/state/state_store_test.go

```diff
@@ -5089,277 +5089,220 @@ func TestStateStore_UpdateAllocsFromClient(t *testing.T) {
 	watches := make([]memdb.WatchSet, 8)
 	for i := 0; i < 8; i++ {
 		watches[i] = memdb.NewWatchSet()
 	}
-	if _, err := state.AllocByID(watches[0], alloc1.ID); err != nil {
-		t.Fatalf("bad: %v", err)
-	}
-	if _, err := state.AllocByID(watches[1], alloc2.ID); err != nil {
-		t.Fatalf("bad: %v", err)
-	}
-	if _, err := state.AllocsByEval(watches[2], alloc1.EvalID); err != nil {
-		t.Fatalf("bad: %v", err)
-	}
-	if _, err := state.AllocsByEval(watches[3], alloc2.EvalID); err != nil {
-		t.Fatalf("bad: %v", err)
-	}
-	if _, err := state.AllocsByJob(watches[4], alloc1.Namespace, alloc1.JobID, false); err != nil {
-		t.Fatalf("bad: %v", err)
-	}
-	if _, err := state.AllocsByJob(watches[5], alloc2.Namespace, alloc2.JobID, false); err != nil {
-		t.Fatalf("bad: %v", err)
-	}
-	if _, err := state.AllocsByNode(watches[6], alloc1.NodeID); err != nil {
-		t.Fatalf("bad: %v", err)
-	}
-	if _, err := state.AllocsByNode(watches[7], alloc2.NodeID); err != nil {
-		t.Fatalf("bad: %v", err)
-	}
+	_, err := state.AllocByID(watches[0], alloc1.ID)
+	must.NoError(t, err)
+	_, err = state.AllocByID(watches[1], alloc2.ID)
+	must.NoError(t, err)
+
+	_, err = state.AllocsByEval(watches[2], alloc1.EvalID)
+	must.NoError(t, err)
+	_, err = state.AllocsByEval(watches[3], alloc2.EvalID)
+	must.NoError(t, err)
+
+	_, err = state.AllocsByJob(watches[4], alloc1.Namespace, alloc1.JobID, false)
+	must.NoError(t, err)
+	_, err = state.AllocsByJob(watches[5], alloc2.Namespace, alloc2.JobID, false)
+	must.NoError(t, err)
+
+	_, err = state.AllocsByNode(watches[6], alloc1.NodeID)
+	must.NoError(t, err)
+	_, err = state.AllocsByNode(watches[7], alloc2.NodeID)
+	must.NoError(t, err)
 
 	// Create the delta updates
 	ts := map[string]*structs.TaskState{"web": {State: structs.TaskStatePending}}
 	update := &structs.Allocation{
 		ID:           alloc1.ID,
+		NodeID:       alloc1.NodeID,
 		ClientStatus: structs.AllocClientStatusFailed,
 		TaskStates:   ts,
 		JobID:        alloc1.JobID,
 		TaskGroup:    alloc1.TaskGroup,
 	}
 	update2 := &structs.Allocation{
 		ID:           alloc2.ID,
+		NodeID:       alloc2.NodeID,
 		ClientStatus: structs.AllocClientStatusRunning,
 		TaskStates:   ts,
 		JobID:        alloc2.JobID,
 		TaskGroup:    alloc2.TaskGroup,
 	}
 
 	err = state.UpdateAllocsFromClient(structs.MsgTypeTestSetup, 1001, []*structs.Allocation{update, update2})
-	if err != nil {
-		t.Fatalf("err: %v", err)
-	}
+	must.NoError(t, err)
 
-	for i, ws := range watches {
-		if !watchFired(ws) {
-			t.Fatalf("bad %d", i)
-		}
+	for _, ws := range watches {
+		must.True(t, watchFired(ws))
 	}
 
 	ws := memdb.NewWatchSet()
 	out, err := state.AllocByID(ws, alloc1.ID)
-	if err != nil {
-		t.Fatalf("err: %v", err)
-	}
+	must.NoError(t, err)
 
 	alloc1.CreateIndex = 1000
 	alloc1.ModifyIndex = 1001
 	alloc1.TaskStates = ts
 	alloc1.ClientStatus = structs.AllocClientStatusFailed
-	if !reflect.DeepEqual(alloc1, out) {
-		t.Fatalf("bad: %#v %#v", alloc1, out)
-	}
+	must.Eq(t, alloc1, out)
```

Comment:
@shoenig is `must.Eq` equivalent to `reflect.DeepEqual`?

Reply:
It uses `cmp.Equal` under the hood, which can be configured with `cmp.Options` if you want

https://github.com/google/go-cmp

Reply:
Ah thanks!

> Unlike reflect.DeepEqual, unexported fields are not compared by default; they result in panics unless suppressed by using an Ignore option (see cmpopts.IgnoreUnexported) or explicitly compared using the AllowUnexported option.

I think we're OK in this case.